app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -1284,7 +1610,45 @@ private void doSave() {
 
         db.setLoyaltyCardGroups(loyaltyCardId, selectedGroups);
 
+        SharedPreferences sharedPreferences = getSharedPreferences("protect.card_locker.dateTime", Context.MODE_PRIVATE);
+        SharedPreferences.Editor editor = sharedPreferences.edit();
+
+        if (showRemindersLayout.getVisibility()==View.VISIBLE && hasReminderChanged ) {
+
+            Log.i(TAG, "doSave: noOfDays " + days);
+            Log.i(TAG, "doSave: hour " + hour);
+            Log.i(TAG, "doSave: minute " + minute);
+            Log.i(TAG, "doSave: formattedDate " + formattedDate);
+
+            editor.putInt(loyaltyCardId + "noOfDays", days);
+            editor.putString(loyaltyCardId + "formattedDate", formattedDate);
+            editor.putInt(loyaltyCardId + "Hour", hour);
+            editor.putInt(loyaltyCardId + "Minute", minute);
+            scheduleNotification();
+
+        }
+        if(showRemindersLayout.getVisibility() != View.VISIBLE) {
+            
+            editor.putInt(loyaltyCardId+"noOfDays",0);
+            Log.i(TAG, "doSave: Alarm not saved");
+            deleteAlarm();
+        }
+
+        editor.apply();
```

Comment:
This needs to be moved to the database, but that is for later and I promised to help with that :)